src/components/MapTiff/MapTiff.tsx

```diff
@@ -167,9 +181,10 @@ const MapTiff = ({
           symbolLayer.id,
         );
       }
+      await cacheMapData();
```

Comment:
```suggestion
      cacheMapData();
```
i think we could leave this asynchronous too. no need to wait for all data be reached before we release one visualization from its loading state.